packages/cli/src/util/on-keypress.ts

```diff
@@ -20,7 +20,9 @@ export default function onKeypress(handleKeyPress: (evt: KeyboardEvent, controls
     });
 
     readline.emitKeypressEvents(process.stdin, rl);
-    process.stdin.setRawMode(true);
+    if (process.stdin.isTTY) {
+      process.stdin.setRawMode(true);
+    }
```

Comment:
This fixes a bug where the `cannon run` command is not working on non tty terminals, e.g. CI or `child_process.spawn`